RubberduckTests/Inspections/UnreachableCaseInspectionTests.cs

```diff
@@ -0,0 +1,2398 @@
+        [TestCase("-1.0_>_0.0?Currency", "False", "Boolean")]
+        [TestCase("-1_<_0", "True", "Boolean")]
+        [TestCase("-1.0_<_0.0?Single", "True", "Boolean")]
+        [Category("Inspections")]
+        public void UciUnit_RelationalOp(string input, string expected, string typeName)
+        {
+            GetBinaryOpValues(input, out IParseTreeValue LHS, out IParseTreeValue RHS, out string opSymbol);
+            var result = Calculator.Evaluate(LHS, RHS, opSymbol);
+            Assert.AreEqual(expected, result.ValueText);
+        }
+
+        [TestCase("False", "False")]
+        [TestCase("True", "True")]
+        [TestCase("-1", "True")]
+        [TestCase("x < 5.55", "x < 5.55")]
+        [Category("Inspections")]
+        public void UciUnit_ConvertToBoolText(string input, string expected)
+        {
+            var result = ValueFactory.Create(input, Tokens.Boolean);
+            Assert.IsNotNull(result, $"Type conversion to {Tokens.Boolean} return null interface");
+            Assert.AreEqual(expected, result.ValueText);
+        }
+
+        [TestCase("Yahoo", "Long")]
+        [TestCase("Yahoo", "Double")]
+        [TestCase("Yahoo", "Boolean")]
+        [Category("Inspections")]
+        public void UciUnit_ConvertToType(string input, string convertToTypeName)
+        {
+            var result =ValueFactory.Create(input, convertToTypeName);
+            Assert.IsNotNull(result, $"Type conversion to {convertToTypeName} return null interface");
+            Assert.AreEqual("Yahoo", result.ValueText);
+        }
+
+        [TestCase("NaN", "String")]
+        [Category("Inspections")]
+        public void UciUnit_ConvertToNanType(string input, string convertToTypeName)
+        {
+            var result = ValueFactory.Create(input, convertToTypeName);
+            Assert.IsNotNull(result, $"Type conversion to {convertToTypeName} return null interface");
+            Assert.AreEqual("NaN", result.ValueText);
+        }
+
+        [TestCase("10.51_*_11.2?Currency", "117.712", "Currency")]
+        [TestCase("10.51?Currency_*_11.2", "117.712", "Currency")]
+        [Category("Inspections")]
+        public void UciUnit_MathOpCurrency(string operands, string expected, string typeName)
+        {
+            var result = TestBinaryOp(MathSymbols.MULTIPLY, operands, expected, typeName);
+            Assert.AreEqual(typeName, result.TypeName);
+        }
+
+        [TestCase("10.51?Long_*_11.2", "123.2", "Double")]
+        [TestCase("10.51?Integer_*_11.2", "123.2", "Double")]
+        [TestCase("10.51?Byte_*_11.2", "123.2", "Double")]
+        [TestCase("10.51?Double_*_11.2", "117.712", "Double")]
+        [TestCase("10_*_11.2", "112", "Double")]
+        [TestCase("11.2_*_10", "112", "Long")]
+        [TestCase("10.51_*_11.2", "117.712", "Double")]
+        [TestCase("10.51?Single_*_11.2?Single", "117.712", "Single")]
+        [TestCase("10.51?Currency_*_11.2?Currency", "117.712", "Single")]
```

Comment:
Why do we expect a `Single` from the expression that multiplies two `Currency` together?

Reply:
When the type is different than the expected result-type of the operation, the test is typically checking that we can convert the results to another type.  This is an important capability because a Select Case Statement can be evaluating a `Long` SelectExpression but one of the Case Statements is math using `Doubles`.  In the end, the `Double`-math Case Statement result needs to be converted to `Long` in order to be properly evaluated against the Select Expression.

Reply:
Great - that makes more sense. May I suggest that the `typeName` be renamed as `selectExpressionTypeName` to make this more clearer that we are also testing the conversion to the data type of the expression, not the resulting evaluation?